client/signup/jetpack-connect/index.jsx

```diff
@@ -199,6 +199,10 @@ class JetpackConnectMain extends Component {
 			return false;
 		}
 
+		if ( this.checkProperty( 'isJetpackConnected' ) ) {
```

Comment:
I think something is a bit off here. Running this PR, I get unexpected behavior trying to link an additional user to an already-connected site. I get the notice and I'm not allowed to connect.